packages/compose/src/hooks/use-media-query/index.js

```diff
@@ -10,28 +28,30 @@ import { useState, useEffect } from '@wordpress/element';
  * @return {boolean} return value of the media query.
  */
 export default function useMediaQuery( query ) {
-	const [ match, setMatch ] = useState(
-		() =>
-			!! (
-				query &&
-				typeof window !== 'undefined' &&
-				window.matchMedia( query ).matches
-			)
+	const subscribe = useCallback(
+		/** @type {(onStoreChange: () => void) => () => void} */
+		( onStoreChange ) => {
+			const mediaQueryList = getMediaQueryList( query );
+			if ( ! mediaQueryList ) {
+				return () => {};
+			}
+
+			mediaQueryList.addEventListener( 'change', onStoreChange );
+			return () => {
+				mediaQueryList.removeEventListener( 'change', onStoreChange );
+			};
+		},
+		[ query ]
 	);
 
-	useEffect( () => {
-		if ( ! query ) {
-			return;
-		}
-		const updateMatch = () =>
-			setMatch( window.matchMedia( query ).matches );
-		updateMatch();
-		const list = window.matchMedia( query );
-		list.addListener( updateMatch );
-		return () => {
-			list.removeListener( updateMatch );
-		};
-	}, [ query ] );
+	const matches = useSyncExternalStore(
+		subscribe,
+		() => {
+			const mediaQueryList = getMediaQueryList( query );
+			return mediaQueryList?.matches ?? false;
+		},
+		() => false
+	);
 
-	return !! query && match;
+	return !! ( query && matches );
```

Comment:
This can be just `return matches` because the `getValue` callback already returns `false` when the `query` is null-ish.